src/app/components/line-dda-result/line-dda-result.component.ts

```diff
@@ -1,20 +1,31 @@
-import { Component, Input, SimpleChanges } from '@angular/core';
+import { Component, Input, SimpleChanges, Output, EventEmitter } from '@angular/core';
 import { CoordinatesService } from '../../services/coordinates.service';
 import { Point, ViewPort, NormalizedRange } from '../../types/coordinates';
-import { DdaMetadata } from '../../types/lines';
+import { DdaMetadata, DdaFixValue } from '../../types/lines';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
 
 @Component({
     selector: 'app-line-dda-result',
     templateUrl: './line-dda-result.component.html',
     styleUrls: ['./line-dda-result.component.scss'],
 })
+
 export class LineDdaResultComponent {
     @Input()
     public metadata: DdaMetadata;
 
     @Input()
     public point: Point;
 
+    @Output()
+    public onDrawLine: EventEmitter<DdaFixValue>;
+
+    constructor(private readonly fb: FormBuilder, private readonly coordinateService: CoordinatesService) {
```

Comment:
Put constructor after the definitions of class members and group your class members by accessibility: public, private, etc.

Reply:
Done